### PR TITLE
Fix rmilter socket path, add deprecation notice

### PR DESCRIPTION
**Pull request by avnik**
###### Motivation for this change

Putting socket and pid file in same workdir cause problem, so I moved socket out of workdir.
Also as rmilter is no more recomended (use built-in rspamd milter instead), deprecation notice was added.

/cc @samueldr this commits suggested for release-18.09 branch

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by joachifm** [comment]
See also https://github.com/NixOS/nixpkgs/pull/46637

File: nixos/modules/services/mail/rmilter.nix [hunk 1]
```diff
@@ -193,6 +193,9 @@ in
   config = mkMerge [
 
     (mkIf cfg.enable {
+      warnings = [
+        ''`config.serivces.rmilter' is deprecated, `rmilter' deprecated and unsupported by upstream, and would be removed from next releases ''
```
Comment (Mic92) [inline nixos/modules/services/mail/rmilter.nix]: `config.services.rmilter` and `will be removed`.

**Comment by avnik** [comment]
On Sun, Sep 23, 2018 at 06:43:43AM -0700, Jörg Thalheim wrote:

> > @@ -193,6 +193,9 @@ in
>    config = mkMerge [
>  
>      (mkIf cfg.enable {
> +      warnings = [
> +        ''`config.serivces.rmilter' is deprecated, `rmilter' deprecated and unsupported by upstream, and would be removed from next releases ''
> 
> `config.services.rmilter` and `will be removed`.
> 

@Mic92 thank you. Fixed, and added suggestion to use rspamd builtin
milter as well.

**Comment by Mic92** [comment]
If you are going to remove rmilter before the 19.03 release, we should backport both commits right?

**Comment by avnik** [comment]
On Sun, Sep 23, 2018 at 05:15:56PM -0700, Jörg Thalheim wrote:

> If you are going to remove rmilter before the 19.03 release, we should backport both commits right?
> 

Thats plan.

If you suggest a slower upgrade path, we can backport only first commit
(with socket path bugfix), announce deprecation in post 18.09, and
remove it right after 19.03 branch.

But rmilter itself was added only as requirement to operate rspamd (0.6
was unable operate as milter directly, but 0.7+ can, and rmilter can be
avoided). So I suggest to all users switch to builtin rspamd's milter.

**Comment by Mic92** [comment]
backport: 4307b377819 and 1a159fe86ba

**Pull request merged**